File: Spotify-ObjC/Spotify-ObjC/SAArtistViewController.m
//
//  SAArtistViewController.m
//  Spotify-ObjC
//
//  Created by Kevin Zeckser on 8/3/16.
//  Copyright © 2016 Kevin Zeckser. All rights reserved.
//

#import "SAArtistViewController.h"

@interface SAArtistViewController ()

@property (weak, nonatomic) IBOutlet UIImageView *artistForegroundImage;
@property (weak, nonatomic) IBOutlet UIImageView *artistBackgroundImage;
@property (weak, nonatomic) IBOutlet UILabel *descriptionLabel;

@end

@implementation SAArtistViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self configureVC];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)configureVC {
    [self configureViews];
    [self configureOutlets];
}

- (void)configureViews {
    UIColor *backgroundColor = [UIColor blackColor];
    UIColor *textColor = [UIColor whiteColor];
    self.view.backgroundColor = backgroundColor;
    self.navigationController.navigationBar.barTintColor = backgroundColor;
    self.navigationController.navigationBar.tintColor = textColor;
    [self.navigationController.navigationBar setTitleTextAttributes: @{NSForegroundColorAttributeName: textColor}];
    
    self.artistForegroundImage.layer.cornerRadius = self.artistForegroundImage.bounds.size.width / 2;
    self.artistForegroundImage.clipsToBounds = true;
    
    UIBlurEffect *blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleDark];
    UIVisualEffectView *blurEffectView = [[UIVisualEffectView alloc] initWithEffect:blurEffect];
    blurEffectView.frame = self.artistBackgroundImage.bounds;
    blurEffectView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    [self.artistBackgroundImage addSubview: blurEffectView];
    
    self.descriptionLabel.textColor = textColor;
}

- (void)configureOutlets {
    if (self.detailArtist) {
        UIImage *artistImage = [self loadImage:self.detailArtist.image];
        self.artistForegroundImage.image = artistImage;
        self.artistBackgroundImage.image = artistImage;
        self.descriptionLabel.text = self.detailArtist.artistDescription;
    }
}

- (UIImage *)loadImage:(NSString *)url {
    if ([url isEqualToString:@""]) {
        return [UIImage imageNamed:@"default_artist_image"];
    } else {
    NSURL *imgURL = [NSURL URLWithString: url];
    NSData *imgData = [NSData dataWithContentsOfURL:imgURL];
    return [UIImage imageWithData:imgData];
    }
}

@end

File: SpotifyArtistsViewer/SAArtistViewController.swift
//
//  SAArtistViewController.swift
//  SpotifyArtistsViewer
//
//  Created by Kevin Zeckser on 6/6/16.
//  Copyright © 2016 Kevin Zeckser. All rights reserved.
//

import UIKit

class SAArtistViewController: UIViewController {
    
    // MARK: - Properties & Outlets
    
    @IBOutlet weak private var artistImageView: UIImageView!
    @IBOutlet weak var descriptionLabel: UILabel!
    var url :String?
    var detailArtist: SAArtist?
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.navigationController?.navigationBar.barTintColor = UIColor.blackColor()
        self.navigationController?.navigationBar.tintColor = UIColor.whiteColor()
        self.navigationController?.navigationBar.titleTextAttributes = [NSForegroundColorAttributeName : UIColor.whiteColor()]
        configureView()
    }

    // MARK: - Configure View Functions
    
    private func loadImage(url :String) -> UIImage? {
        guard
            let imgURL = NSURL(string: url),
            let imgData = NSData(contentsOfURL: imgURL)
            else { return nil }
        return UIImage(data: imgData)
        
    }
    
    private func configureView() {
        if let detailArtist = detailArtist {
            artistImageView.image = loadImage(detailArtist.image!)
            artistImageView.layer.cornerRadius = artistImageView.bounds.size.width / 2
            artistImageView.clipsToBounds = true
            descriptionLabel.text = detailArtist.description
        }
    }
}
